lib/src/features/dashboard_page.dart

```diff
@@ -11,35 +11,63 @@ class DashboardPage extends StatefulWidget {
   State<DashboardPage> createState() => _DashboardPageState();
 }
 
-class _DashboardPageState extends State<DashboardPage> {
+class _DashboardPageState extends State<DashboardPage> with SingleTickerProviderStateMixin{
+
+  late TabController tabController;
+
+  @override
+  void initState(){
+    super.initState();
+    tabController = TabController(length: 4, vsync: this, initialIndex: 0);
+  }
+
   @override
   Widget build(BuildContext context) {
-    return DefaultTabController(
-        length: 4,
-        child: Scaffold(
-        appBar: AppBar(
+    return Scaffold(
+      appBar: AppBar(
           title: Text("Whatsapp"),
+          actions: [
+            IconButton(onPressed: () {}, icon: Icon(Icons.search)),
+            PopupMenuButton<String>(
+                itemBuilder: (BuildContext context) {
+                  return [
+                    PopupMenuItem(child: Text("New group"), value: "New group",),
+                    PopupMenuItem(child: Text("New broadcast"), value: "New broadcast",),
```

Comment:
You can create separate class file for Strings and create variables for them. You can access them here